### PR TITLE
make surveyor workflow functional

### PR DESCRIPTION
**Pull request by mattdelsordo**
done:
1. assigning/removing assignments to surveyors
2. surveyors can see only their assignments
3. clicking an address takes you to the survey
4. removed in-between pages to streamline things

still needed:
1. the completed field on homes
2. geospatial data on homes

### CODE REVIEW

File: backend/app/controllers/assignments_surveyors_controller.rb
Comment (mattdelsordo) [inline backend/app/controllers/assignments_surveyors_controller.rb:1]: Could not for the life of me figure out how to modify the assignment-surveyor relationship so I ended up making a new controller for it.

File: frontend/front/src/pages/Admin/nav/Nav.js [hunk 1]
```diff
@@ -11,11 +11,14 @@ const Nav = () => {
...
         <Toolbar>
-          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
+          <Box sx={{ flexGrow: 1 }}>
             <Button color="inherit" component={Link} to="">
-              <Typography variant="h6">ADMIN</Typography>
+              <Typography variant="h6">DASHBOARD</Typography>
             </Button>
-          </Typography>
+            <Button color="inherit" component={Link} to="/surveyor/dashboard">
+              <Typography variant="h7">SURVEYOR MODE</Typography>
```
Comment (mattdelsordo) [inline frontend/front/src/pages/Admin/nav/Nav.js:19]: Admins can now click "Surveyor Mode" in the nav to be treated like a surveyor

File: frontend/front/src/pages/Surveyor/dashboard/AssignmentUnit.js [hunk 2]
```diff
@@ -1,121 +1,90 @@
...
+  return 0;
+};
 
-const AssignmentUnit = ({ data }) => {
-  const dispatch = useDispatch();
+const generateGoogleMap = (checked) => {
+  const sortedCheck = [...checked].sort(compareArrayByOrder);
 
-  const navigate = useNavigate();
+  window.open(generateMapsLink(sortedCheck, "_blank"));
```
Comment (mattdelsordo) [inline frontend/front/src/pages/Surveyor/dashboard/AssignmentUnit.js:22]: Clicking "Generate Map" now links directly to google maps

File: frontend/front/src/pages/Surveyor/nav/Nav.js [hunk 3]
```diff
@@ -22,18 +30,26 @@ const Nav = () => {
                 DASHBOARD
               </Typography>
             </Button>
+
+            {userIsAdmin && (
```
Comment (mattdelsordo) [inline frontend/front/src/pages/Surveyor/nav/Nav.js:34]: If the surveyor is also an admin, this links to the admin page

File: frontend/front/src/pages/Admin/nav/Nav.js (hunk 1 above)
Comment (Rando-cal) [reply frontend/front/src/pages/Admin/nav/Nav.js:19]: What Surveyor does it mimic?

Comment (mattdelsordo) [reply frontend/front/src/pages/Admin/nav/Nav.js:19]: It just behaves as if the admin user was a surveyor user

**Review by Rando-cal — APPROVED** [review]

**Pull request merged**